### ISSUE TITLE
NPX swallowing stack trace

### ISSUE THREAD

**Issue by dyst5422** [npm/npx#36]
Stack trace is consumed by npx here https://github.com/npm/npx/blob/26a839472058a1817f24d8ca41a74f41bc4f98ab/index.js#L113

As a user, this is pretty unacceptable.

**Comment by crystalneth** [comment]
This is really bad for development! I can understand this behavior for certain user commands, however when using npx for development or debugging, e.g. `npx ts-node ...`, this is a  blocker.

The workaround seems to be installing ts-node globally or otherwise avoiding npx, which obviously is problematic in other ways.

**Comment by dyst5422** [comment]
@crystalneth I cannot imagine a single scenario where I would want errors to be suppressed. Even user commands should be forwarding errors

**Comment by crystalneth** [comment]
Yes you are probably correct, as if an application wants to print a user error rather than a stack dump, the application should catch the error and report a clean user-oriented error.

Hiding errors the way npx is doing is a programming 101 fail. Hopefully this can be fixed, as it is a part of the official npm module.

Installing ts-node globally worked for my case, although it isn't ideal.

**Comment by Taytay** [comment]
If `npx` maintainers would prefer to keep this to a 1-line error for backwards compat, perhaps there could be a verbose-error mode? I just ran into this as well and was surprised not to see where my ts-node project was failing.

**Comment by Swivelgames** [comment]
I agree. This is a major problem for development. Without stack traces, we've been running around in circles on an issue. It took quite some time just to figure out why the stack traces were gone to begin with.

**Comment by tomstaijen** [comment]
There's a workaround that has been working for me, to run `npx -s sh <program>` instead of `npx <program>`.

**Comment by ccurrie-amzn** [comment]
Tagging zkat/npx#188 to help others like me who found this reported there first.
